### DISCUSSION TITLE
Upgrading from 16.04 to 2.3.10 Issues

### DISCUSSION THREAD

**Discussion by femaven** [Security-Onion-Solutions/securityonion#2073] [Unsupported Versions]
My thread in #1968 got hijacked so opening a new discussion.
The Ubuntu distribution completes successfully. I'm just having issues with the upgrade script.
Exact version. ex. 2.3.10
Install source. Did you install from our Security Onion ISO image or did you perform a network installation? network install from 16.04
If network install, did you install from CentOS 7 or Ubuntu 18.04? Ubuntu 18.04
Install type. ex. eval, standalone, forward node, search node, etc - Eval
Does so-status show all services running? The only service that's running is docker-registry.
Do you get any failures when you run salt-call state.highstate from the node? Yes
See highstate.log attached
See sosetup.log attached
[highstate.log](https://github.com/Security-Onion-Solutions/securityonion/files/5590914/highstate.log)
[sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/5590916/sosetup.log)

**Comment by weslambert** [comment]
Seems like `apt` isn't happy, and there are broken packages.  What happens if you try to run an `apt update` or `apt install -f`?

**Reply by femaven** [reply]
Could it be an issue with my salt repo config? this is the output that I get:

```
root@securityonion:~# apt update
Hit:1 http://us.archive.ubuntu.com/ubuntu bionic InRelease
Hit:2 http://us.archive.ubuntu.com/ubuntu bionic-updates InRelease
Hit:3 https://download.docker.com/linux/ubuntu bionic InRelease
Hit:4 https://packages.wazuh.com/3.x/apt stable InRelease
Hit:5 https://repo.saltstack.com/py3/ubuntu/18.04/amd64/archive/3002.2 bionic InRelease
Hit:6 http://us.archive.ubuntu.com/ubuntu bionic-backports InRelease
Hit:7 http://security.ubuntu.com/ubuntu bionic-security InRelease
Reading package lists... Done
Building dependency tree
Reading state information... Done
6 packages can be upgraded. Run 'apt list --upgradable' to see them.
N: Skipping acquire of configured file 'main/binary-i386/Packages' as repository 'https://repo.saltstack.com/py3/ubuntu/18.04/amd64/archive/3002.2 bionic InRelease' doesn't support architecture 'i386'
```

```
root@securityonion:~# apt install -f
Reading package lists... Done
Building dependency tree
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 6 not upgraded.
root@securityonion:~#
```

**Comment by femaven** [comment]
when I changed the salt repo to amd64 using the below config I don't get an error with _apt-get update_
deb [arch=amd64] http://repo.saltstack.com/py3/ubuntu/18.04/amd64/latest bionic main
but when I rerun setup in 2.3.10, the error returns. It seems like there's an issue with the install script. I updated line 1545 in so-functions (saltify) to                                 
_echo "deb [arch=amd64] http://repo.saltstack.com$py_ver_url_path/ubuntu/$ubuntu_version/amd64/archive/3002.2 $OSVER main" > /etc/apt/sources.list.d/saltstack.list 2>> "$setup_log"_

And now I get the following:
[sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/5593327/sosetup.log)
So slow progress.
